sklearn/utils/tests/test_testing.py

```diff
@@ -96,10 +101,53 @@ def _no_raise():
                   "test", _no_raise)
 
 
+def test_assert_same_not_same_model():
+    X1, y1 = make_blobs(n_samples=200, n_features=5, center_box=(-200, -150),
+                        centers=2, random_state=0)
+    X2, y2 = make_blobs(n_samples=100, n_features=5, center_box=(-1, 1),
+                        centers=3, random_state=1)
+    X3, y3 = make_blobs(n_samples=50, n_features=5, center_box=(-100, -50),
+                        centers=4, random_state=2)
+
+    # Checking both non-transductive and transductive algorithms
+    # By testing for transductive algorithms we also eventually test
+    # the assert_fitted_attributes_equal helper.
+    for Estimator in (LinearSVC, KMeans):
+        assert_same_model(X3, Estimator(random_state=0).fit(X1, y1),
+                          Estimator(random_state=0).fit(X1, y1))
+        assert_raises(AssertionError, assert_not_same_model, X3,
+                      Estimator(random_state=0).fit(X1, y1),
+                      Estimator(random_state=0).fit(X1, y1))
+        assert_raises(AssertionError, assert_same_model, X3,
+                      Estimator(random_state=0).fit(X1, y1),
+                      Estimator(random_state=0).fit(X2, y2))
+        assert_not_same_model(X3, Estimator(random_state=0).fit(X1, y1),
+                              Estimator(random_state=0).fit(X2, y2))
+
+
+def test_qda_same_model():
+    # NRT to make sure the rotations_ attribute is correctly compared
```

Comment:
Sorry for the lack of clarity in the comment... @amueller had pointed out in one comment that QDA was not correctly compared by the previous implementation of the `assert_same_model`. That was because rotations was a list of numpy arrays and so I thought it would be worthwhile to add a NRT for QDA alone!

Will update the comment to make it more clear!